src/main/ionic/peers/src/components/edit-profile-modal/edit-profile-modal.ts

```diff
@@ -0,0 +1,56 @@
+import { Component } from '@angular/core';
+import {NavController, NavParams, Platform, ViewController} from 'ionic-angular';
+import {UserService} from "../../services/user-service";
+import {AppState} from "../../states/app-state";
+import { MyProfile } from "../../pages/my-profile/my-profile";
+
+@Component({
+  selector: 'edit-profile-modal',
+  templateUrl: 'edit-profile-modal.html'
+})
+export class EditProfileModal {
+  data: any;
+  userInfo = {
+    firstName: '',
+    lastName: '',
+    level: '',
+    major: '',
+    interests: '',
+    bio: '',
+  };
+
+  private myProfile: MyProfile;
+
+  constructor(public platform: Platform,
+              public params: NavParams,
+              public viewCtrl: ViewController,
+              private userService: UserService,
+              private appState: AppState,
+              private nav: NavController) {
+    this.userInfo.firstName = params.get("firstName");
+    this.userInfo.lastName = params.get("lastName");
+    this.userInfo.level = params.get("level");
+    this.userInfo.major = params.get("major");
+    this.userInfo.interests = params.get("interests");
+    this.userInfo.bio = params.get("bio");
+    this.myProfile = params.get("myProfile");
+  }
+
+  public save() {
+    this.viewCtrl.dismiss(null);
+
+  }
+
+  public dismiss() {
+    this.viewCtrl.dismiss(null);
+  }
+
+  public submitEditUser() {
+    this.appState.setIsLoading(true);
+    this.userService.updateUser(this.userInfo).then(res => {
+      this.appState.setIsLoading(false);
+      this.myProfile.updateView(this.userInfo);
```

Comment:
Maybe we should create a UserState class just like CardsState. Because this current implementation leads to the risk that user info is out of sync throughout different pages in our application.

Reply:
You mean we want to keep the user status locally? For now, everything related to user profile is gotten from the database

Reply:
Ideally yes, we get the user information upon login and update it on edit. All pages that use information from the current user can subscribe to it so they will never be out of sync.
For instance say you change your interests and then go back to the home page. The user info object containing the interests that we use to request cards from the backend is not updated.